src/script/interpreter.cpp

```diff
@@ -170,6 +170,13 @@ bool static IsValidSignatureEncoding(const std::vector<unsigned char> &sig) {
     return true;
 }
 
+uint32_t static GetHashType(const valtype &vchSig) {
+    if (vchSig.size() == 0)
+        return 0;
+    // check IsValidSignatureEncoding()'s comment for vchSig format
+    return vchSig[vchSig.size() - 1];
+}
+
```

Comment:
`return (vchSig.size() == 0) ? 0 : vchSig[vchSig.size() - 1]` is less code to maintain, just a suggestion 😺